advancedlog/legacy_logger.go

```diff
@@ -0,0 +1,53 @@
+package logger
+
+import (
+	logutils "github.com/bitrise-io/go-utils/v2/log"
+	"strings"
+)
+
+type legacyLogger struct {
+	debugLogEnabled bool
+	logger          logutils.Logger
+}
+
+func newLegacyLogger(logger logutils.Logger) SimplifiedLogger {
+	return &legacyLogger{
+		debugLogEnabled: false,
+		logger:          logger,
+	}
+}
+
+func (l *legacyLogger) EnableDebugLog(enabled bool) {
+	l.logger.EnableDebugLog(enabled)
+	l.debugLogEnabled = enabled
+}
+
+func (l *legacyLogger) IsDebugLogEnabled() bool {
+	return l.debugLogEnabled
+}
+
+func (l *legacyLogger) LogMessage(producer Producer, level Level, message string) {
+	if l.debugLogEnabled == false && level == DebugLevel {
+		return
+	}
+
+	// It is needed to trim the newline char from the end because the wrapped logger will automatically add one
+	if message != "\n" {
```

Comment:
IMO we should use log functions that add newline to the end of the message and we shouldn't modify the original message. With these 2 I think:
- we should remove the newlines on the client side, where it is not needed (where we use old log functions and used a logging method that doesn't add newline at the end, but we added one)
- we should remove this workaround from here

Reply:
Okay, you convinced me. Initially I thought that this is not a big deal but if we really want to preserve the original message (like when the developer put a newline at the end on purpose) then we need to remove this.